draft-specifications/Locators.md

```diff
@@ -0,0 +1,137 @@
+# Warning: This is a draft specification likely to undergo significant change
+
+This draft specification has been initially written based on Spritely Goblin's
+implementation of as a base for the [OCapN pre-standardization
+group](https://ocapn.org) to work from. Over time this document will change,
+likely significantly as the group  converges on the design of OCapN. If you're
+interested in being part of that  work, please join!
+
+Authors: Jessica Tallon, Christine Lemmer-Webber & The OCapN Pre-standardization
+Group.
+
+# [Introduction](#introduction)
+
+OCapN Locators used to identify OCapN capable nodes or objects available on a
+specific node. They can be used in band as Syrup encoded data or out of band
+when bootstrapping a connection as URIs.
+
+These locators are agnostic to the netlayer that the node or object is located
+on, it encodes the transport  protocol name, key and other additional data which
+would be used by any given  netlayer to reach the node.
+
+# Additional Documents
+
+This document does not stand alone, it relies upon multiple other documents
+which together build up OCapN  (Object Capability Network) specifications.
+
+This specification uses the following other specifications:
+
+-   [Syrup](): The serialization format used to encode the in band
+    representation of the locator. 
+-   [OCapN Netlayers](): Specification to open a secure communication channel
+    between two sessions, often on  different networks.
+-   [RFC3986: Uniform Resource Identifier (URI): Generic
+    Syntax](https://www.rfc-editor.org/rfc/rfc3986): For encoding URI
+    representations of locations.
+
+# [Node Locator](#ocapn-node)
+
+This identifies an OCapN node, not a specific object. This includes enough
+information to specify which netlayer and provide that netlayer with all of the
+information needed to create a  bidirectional channel to that node.
+
+The node locator include the following pieces of information (more details
+below):
+
+- **Designator**: Usually representing the key, however can be any value
+  determined by the netlayer
```

Comment:
My understanding of @warner’s position is that the designator should be the fingerprint of a public key that is the same regardless of the transport, such that a nodes reachable over multiple transports have the same designator. That would suggest that the transport would be more tightly coupled to the hints.

A useful thought experiment: how would the locator appear for a node that is reachable via Tor and WebSocket over HTTPS, such that the client can choose the transport based on availability of transport clients from their node? My expectation is that the locator would be of the form `( fingerprint, nonce?, hints: ( transport, transport-specific-extra )+ )`.